all_test.go

```diff
@@ -12,18 +13,37 @@ import (
 )
 
 func TestMain(m *testing.M) {
-	setup(m)
+	if err := setup(m); err != nil {
+		teardown(m)
+		log.Fatalf("failed setup: %v", err)
```

Comment:
Friendly log ❤️

Reply:
I've adjusted it to use `log.Printf()` instead.

Reply:
Is that because we wanna make sure to run `teardown`?

Reply:
Yes. If setup fails, it maybe still succeeded partially, so we should run teardown just in case.

Reply:
Fair enough. Thank you.